pkg/v1/builder/command/cli_compile.go

```diff
@@ -239,6 +239,7 @@ func buildPlugin(path string, arch cli.Arch, id string) (plugin, error) {
 
 	var modPath string
 
+	ldflags += " -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Version=" + version + "' "
```

Comment:
That's a good catch; we will definitely use https://github.com/vmware-tanzu/tanzu-framework/blob/main/cli/runtime/buildinfo/buildinfo.go after the refactoring, for now, I will add a comment and also add the Version from  https://github.com/vmware-tanzu/tanzu-framework/blob/main/cli/runtime/buildinfo/buildinfo.go, so that we can avoid bug if in case refactoring ignores adding Version from cli/runtime/buildinfo/buildinfo.go

Reply:
I am not sure we should do this hardcoded string here because it requires change in builder plugin everytime this `buildinfo` gets moved to different location.

Also, this is just general recommendation that plugins use to define their versions but plugin developer can hardcode the plugin version or rely on some other means to define plugin version in the PluginDescriptor. So, assuming this might not be good idea. 

We need to think about passing this type of version is even possible to build a specific plugin or not.

Reply:
@chandrareddyp I don't think we should be hardcoding an ldflag in the code. IMO we should be able to set `buildinfo.Version` by doing something like what's being done in the Makefile - https://github.com/vmware-tanzu/tanzu-framework/blob/main/Makefile#L87-L115

Reply:
> @chandrareddyp I don't think we should be hardcoding an ldflag in the code. IMO we should be able to set `buildinfo.Version` by doing something like what's being done in the Makefile 

The problem with that is that if you just call `tanzu builder cli compile --match package --version v0.1.2` the `--version` flag will be ignored, which seems wrong from a user's point-of-view.  This is related to why @chandrareddyp was asking what this `--version` flag is really meant to do.

> I am not sure we should do this hardcoded string here because it requires change in builder plugin everytime this `buildinfo` gets moved to different location.

True.

> Also, this is just general recommendation that plugins use to define their versions but plugin developer can hardcode the plugin version or rely on some other means to define plugin version in the PluginDescriptor. So, assuming this might not be good idea.
> We need to think about passing this type of version is even possible to build a specific plugin or not.

Yeah, that is tricky.  It comes back to "what is the `--version` flag supposed to do?"

Reply:
> The problem with that is that if you just call tanzu builder cli compile --match package --version v0.1.2 the --version flag will be ignored, which seems wrong from a user's point-of-view.

I was not involved in the development of this plugin/command, however, Based on the help-text for version, it seems it is meant only for core-cli version and not for plugin version. ` --version string       Version of the root cli (required)`

I think, the main issue here is that `--version` flag should not be a required flag as we expect user to just build the plugin without core-cli.

Reply:
The  `--version` flag is not required to compile the plugin, so we have made changes to make `--version` flag as optional.